cmake/parthenonConfig.cmake.in

```diff
@@ -0,0 +1,26 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Favor using the kokkos package that was built with parthenon, if one has not been specified 
+if ( NOT Kokkos_FOUND )
```

Comment:
Not sure why this out `if()` is needed.

Reply:
1. Currently Kokkos is built as a submodule, include(external/Kokkos) so it makes sense to force the user to build with it in that case, I was not sure how cmake handles external libraries. If you use find_package(Kokkos 3) will it also install a config file in the same install dir? 

2. I did know know what the convention is, for third party libraries, if you always want to force the user to use the library that was built with parthenon or not?

Reply:
If you want to force the same library just do something like this: https://github.com/votca/tools/blob/646d1c22564aa3bb524be25dd9a556688f54092d/src/libtools/VOTCA_TOOLSConfig.cmake.in#L11-L17

Reply:
Well, the problem is that when the config gets included by an external package `Kokkos_FOUND` isn't set, the only way to have set already is if the consumer call `find_package(Kokkos)` itself.

Reply:
I removed the `if ( NOT Kokkos_FOUND )` and everything seems OK. Also, I'm using `find_package(Kokkos [...] NO_DEFAULT_PATH)` to (hopefully) enforce specificity to the Kokkos install used when compiling parthenon